public/app/features/dashboard/api/publicDashboardApi.ts

```diff
@@ -69,40 +70,59 @@ export const publicDashboardApi = createApi({
     }),
     createPublicDashboard: builder.mutation<
       PublicDashboard,
-      { dashboard: DashboardModel; payload: Partial<PublicDashboardSettings> }
+      { dashboard: DashboardModel | DashboardScene; payload: Partial<PublicDashboardSettings> }
     >({
       query: (params) => ({
-        url: `/dashboards/uid/${params.dashboard.uid}/public-dashboards`,
+        url: `/dashboards/uid/${
+          params.dashboard instanceof DashboardScene ? params.dashboard.state.uid : params.dashboard.uid
+        }/public-dashboards`,
         method: 'POST',
         data: params.payload,
       }),
```

Comment:
I would make it cleaner

    query: (params) => {
        const dashboardUid =  params.dashboard instanceof DashboardScene ? params.dashboard.state.uid : params.dashboard.uid;
        return {
          url: `/dashboards/uid/${dashboardUid}/public-dashboards`,
          method: 'POST',
          data: params.payload,
        };
      },